### ISSUE TITLE
Structs should have option to generate with zero values rather than Option

### ISSUE THREAD

**Issue by alelazar** [tokio-rs/prost#763]
https://developers.google.com/protocol-buffers/docs/proto3#default

Take a simple example

```
message Coords {
    int32 x = 1;
    int32 y = 2;
}

message Player {
  string uuid = 1;
  Coords coords = 2;
}
```

This will generate with `Player.coords` being an `Option`. This is reasonable, compared to the `has_coords` that most other languages have. However, most other languages would still have the default value in for `coords`. If you know that coords is there and valid, you don't need to mess about. 

`Option` is zero cost in running but can be significant clutter in code if not necessary. I understand that for many people, this is how they want their proto to generate. It would be nice to have the option to remove option and just put the default value if not present.

**Comment by banool** [comment]
I agree this would be an awesome usability win instead of having to `.context("Field unexpectedly asbent")?` on every field.

**Comment by teamplayer3** [comment]
I stumbled across this as well. I don't understand why a filed with a self defined message type is modeled as a rust `Option` and decorated with `optional` in the generated code.

```proto
message Uuid {
    string value = 1;
}

message Player {
    Uuid uuid = 1;
}
```

```rust
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct Player {
    #[prost(message, optional, tag="1")]
    pub uuid: ::core::option::Option<Uuid>,
}
```

When the Player type is created, the uuid could be left out with `None`. But this would be against schema definition where it is required as default. If there isn't another reason I don't see, this must be changed to either don't use `Option` for these fields or use access functions on the fields to set and get them and the unwrapping must be done there.

**Comment by ForsakenHarmony** [comment]
I don't feel like `Option` is ever the right choice for default values, if the optimization is wanted it should be another type

**Comment by mitsuhiko** [comment]
I wonder if an option would be to expand on `.extern_path()` and similar APIs on the builder. Something like `.use_default("name of type")` to make it use `Default::default` rather than wrapping in `Option`.

**Comment by mzabaluev** [comment]
> https://developers.google.com/protocol-buffers/docs/proto3#default

The Protobuf spec actually holds that message-typed fields always have explicit presence; that never changed between proto2/proto3/editions. Custom default values are not available for message fields. Protobuf cannot specify what your application wants to do in case the field is absent, which is always a valid encoding. So whether you use `Option::unwrap_or_default`, `Option::ok_or`, or any of the other ways to get at the possibly missing value is up to the application's needs.

If you want precise control over your data schema, you can write some data domain types with proper encapsulation, invariant enforcement, idiomatic accessors and whatnot, and provide fallible conversions from prost-generated structs. prost-build cannot do it for you, and no amount of customization options will change this.
